gstcefsrc.cc

```diff
@@ -680,6 +696,11 @@ gst_cef_src_class_init (GstCefSrcClass * klass)
           "The URL to display",
           DEFAULT_URL, (GParamFlags) (G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS | G_PARAM_CONSTRUCT)));
 
+  g_object_class_install_property (gobject_class, PROP_GPU,
+    g_param_spec_boolean ("gpu", "gpu",
+          "Enable GPU",
+          FALSE, (GParamFlags) (G_PARAM_READWRITE | G_PARAM_STATIC_STRINGS)));
```

Comment:
Not obvious, but this property will only be used when the element undergoes the READY -> PAUSED state change, so you want to indicate that it is `GST_PARAM_MUTABLE_READY` :)

Reply:
Interesting... gstreamer change properties after it has been started?

Reply:
Well, the property mechanism can be used by whoever is using this element when the element is in any state through the API, when using an element with gst-launch the property will be set when the element is still in the NULL state, but gst-launch is not the sole user of gstreamer's API :) By using `GST_PARAM_MUTABLE_READY` you signal to whoever might call `g_object_set(prop, provalue)` on the element that their changes will not be used until the element cycles back to READY , if it is in a higher state :)